core/src/main/scala/kafka/raft/KafkaMetadataLog.scala

```diff
@@ -96,10 +96,10 @@ final class KafkaMetadataLog private (
   }
 
   private def handleAndConvertLogAppendInfo(appendInfo: internals.log.LogAppendInfo): LogAppendInfo = {
-    if (appendInfo.firstOffset.isPresent())
-      new LogAppendInfo(appendInfo.firstOffset.get().messageOffset, appendInfo.lastOffset)
+    if (appendInfo.firstOffset != UnifiedLog.UnknownOffset)
+      new LogAppendInfo(appendInfo.firstOffset, appendInfo.lastOffset)
     else
-      throw new KafkaException(s"Append failed unexpectedly: ${appendInfo.errorMessage}")
```

Comment:
`errorMessage` would have never been populated in the calling context of `handleAndConvertLogAppendInfo`, so it is just dropped from the log in the PR.